src/Multipart/AbstractUploadManager.php

```diff
@@ -92,8 +93,11 @@ public function promise()
         if ($this->promise) {
             return $this->promise;
         }
+        $exception = $this->client->getApi()->getServiceName() === 's3'
```

Comment:
This is coupling concrete exceptions to the abstract class. This isn't a good way to achieve what you want.

Considering this class is marked as internal, why not make this an argument passed in to the constructor instead?